cfme/fixtures/v2v_fixtures.py

```diff
@@ -0,0 +1,468 @@
+import itertools
+import json
+from collections import namedtuple
+
+import fauxfactory
+import pytest
+from riggerlib import recursive_update
+from widgetastic.utils import partial_match
+
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.fixtures.provider import setup_or_skip
+from cfme.infrastructure.provider.rhevm import RHEVMProvider
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.utils.generators import random_vm_name
+from cfme.utils.log import logger
+from cfme.v2v.infrastructure_mapping import InfrastructureMapping as InfraMapping
+
+
+FormDataVmObj = namedtuple("FormDataVmObj", ["infra_mapping_data", "vm_list"])
+V2vProviders = namedtuple("V2vProviders", ["vmware_provider", "rhv_provider", "osp_provider"])
+
+
+@pytest.fixture(scope="module")
+def v2v_provider_setup(request, source_provider, provider):
+    """ Fixture to setup providers """
+    vmware_provider, rhv_provider, osp_provider = None, None, None
+    for v2v_provider in [source_provider, provider]:
+        if v2v_provider.one_of(VMwareProvider):
+            vmware_provider = v2v_provider
+            setup_or_skip(request, vmware_provider)
+        elif v2v_provider.one_of(RHEVMProvider):
+            rhv_provider = v2v_provider
+            setup_or_skip(request, rhv_provider)
+        elif v2v_provider.one_of(OpenStackProvider):
+            osp_provider = v2v_provider
+            setup_or_skip(request, osp_provider)
+        else:
+            pytest.skip("Provider {} is not a valid provider for v2v tests".format(provider.name))
+    v2v_providers = V2vProviders(vmware_provider=vmware_provider,
+                                 rhv_provider=rhv_provider,
+                                 osp_provider=osp_provider)
+    yield v2v_providers
+    for v2v_provider in v2v_providers:
+        if v2v_provider is not None:
+            v2v_provider.delete_if_exists(cancel=False)
+
+
+@pytest.fixture(scope="module")
+def host_credentials(request, v2v_provider_setup):
+    """ Sets up host credentials for vmware and rhv providers.
+    Args:
+        request:
+        v2v_provider_setup:
+    """
+    v2v_providers = v2v_provider_setup
+    provider_list = [v2v_providers.vmware_provider, v2v_providers.rhv_provider]
+    rhv_hosts = v2v_providers.rhv_provider.hosts.all()
```

Comment:
what if v2v_providers.rhv_provider is `None` in this case, this would throw an `AttributeError`